### ISSUE TITLE
e3.v1 current conditions (for a location)

### ISSUE THREAD

**Issue by janelx** [weather-gov/weather.gov#175]
[Figma link](https://www.figma.com/file/0dSABt9zCqoknaKua6DHmP/NOAA---Weather.gov?type=design&node-id=363-2510&mode=dev)
On pages/layouts: #174. Maybe: #173, #160  

## Supported user stories
- 3.1 … I want to understand that the weather information I'm seeing is relevant to my specific location
- 3.2 … I want to view current conditions

## Support needed
- Weather APIs
- [New icons and mapping](url) 
- Automated text summary ?? weather.gov might have a current solution??

- [x] complete #34  
## Acceptance Criteria

*Not part of this story: search/change location.*

**As a member of the public I can easily understand:**

- what the current weather is for a location
- Use a screen reader to listen/read a quick and easy to understand summary of what is shown on screen

----
### General

- [ ] Responsive
- [ ] USWDS friendly
- [ ] Section 508 compliant

### Display current weather for a single location
[Figma link](https://www.figma.com/file/0dSABt9zCqoknaKua6DHmP/NOAA---Weather.gov?type=design&node-id=363-2510&mode=dev)
![Image](https://github.com/weather-gov/weather.gov/assets/93934808/a280c0b2-ccc4-446d-98a8-67d8c80c571f)
[current conditions api (cc)](https://github.com/weather-gov/weather.gov/blob/main/docs/data.md#current-conditions-observational-data)

### Timestamp

<kbd> ![Image](https://github.com/weather-gov/weather.gov/assets/93934808/3fb0b431-4b1a-4c73-8210-430b20b910e1) </kdb>

- 11:11 AM CDT > [timestamp] 
- gray-cool-50 
- 0.75 rem

### Basic weather information
![Image](https://github.com/weather-gov/weather.gov/assets/93934808/474e1074-9f00-4633-a3aa-ffeb37d5b799)

**Weather icons**
- New NOAA Icons - See [google sheet for mapping](https://docs.google.com/spreadsheets/d/1Er4nW_WA2nu-9GdeoUwT4JkSbfp7B84hUnXa32PQcdk/edit#gid=1809319678)
  - Replace old NWS weather condition icons with corresponding NOAA icon. 
  - Some day/night mapping may use the same icon. 
- Height = 64px
- Width = unconstrained/relative to height
_Notes: Used [Icons API](https://api.weather.gov/icons) and [images in our github](https://github.com/18F/weather.gov/tree/main/docs/icons) to map._

**79 °F**  >  [temperature]
- “79” =  3 rem (let’s try this as scalable instead of fixed)
- °F = 1.5 rem

**Rain showers (high cloud cover)** > [textDescription] of current conditions or icon api [descriptions]
- Rain showers > body

### Additional weather info
![Image](https://github.com/weather-gov/weather.gov/assets/93934808/3e93682f-74bd-4b89-a5a4-c3220f24ebdc)

**Feels like temp**
-“83” = 1.5 rem
- “°F” = 0.75 rem
- “Feels like”  = Gray / 0.75 rem
  - apparentTemperature (gridpoints)
  - calculated  using: cc api > wind chill and cc api > heat index

**Humidity: 55%**
- cc api > relative humidity

**Wind speed and direction**
- cc api > wind speed
- cc api > wind direction
- bonus: arrow icon w/corresponding wind speed

### Summary of immediate forecast (accessibility)
![Image](https://github.com/weather-gov/weather.gov/assets/93934808/8a95b448-1b17-4a20-ab9c-ee777bd6740c)
    - Canned text "The weather in [Nashville, TN] is [rain with high cloud cover]. Temperature is [79 °F]. Humidity is [55%] with [10mph] winds from the [northeast]." bonus - "warmer/cooler than yesterday"
    - Bonus: add text that states, "number of degrees warmer/colder than yesterday". ex: "5 degrees colder than yesterday." We can start w/displaying it in the text to see if this additional detail is useful and potentially evolve the design with testing/feedback in later iterations. 

### Mobile 
- Can we see how it scales for mobile and decide from there the breakpoints?

----

```[tasklist]

### Review: design, function and flow:
- [ ] POs - org alignment and vision review
- [ ] Eng - feasibility review
- [ ] UX - what and how to test

```

```[tasklist]

### Story details
-  [ ] Reviewed designs (must be attached)
-  [ ] Acceptance criteria define (added to this story)

```

---

### Implementation work that needs doing before this is "done:"
- [x] #199
- [x] #200
- [ ] Verify responsiveness (this will be after #189)
- [ ] Run accessibility checks (fully manual this time; also after #189)

**Comment by janelx** [comment]
@greg-does-weather I've updated the gray used to one that is already used in USWDS - 'gray-cool-50'

**Comment by greg-does-weather** [comment]
Can I propose that "translation-friendly" be removed from acceptance criteria for now? This component raises a special challenge, I think, because the only text we control has text we don't control embedded into it:

> The weather in **LOCATION** is **DESCRIPTION**. Temperature is **TEMP**. Humidity is **HUMIDITY** with **WIND** from the **DIRECTION**.

That `DESCRIPTION`, in particular, is in English from the API.

I have a couple thoughts about how we might do it and I'll write them here so I don't forget:
1. Instead of displaying the short descriptions provided by the API, use the text from the API as a key into our own dictionary of expressions. That dictionary of terms could then use our translations.
2. The really tricky part is that the text stored as the translation will need to have tokens in it so we can insert the live data into it. Something like:

   ```
   The weather in :location: is :description:.
   ```

   What's tricky about it is the UX of entering tokenized text like that is probably pretty cruddy.

Anyway, we haven't built any of the infrastructure to support translations yet, so if that's in this issue's acceptance criteria, it's going to linger out here, open for a long time. 😬

**Comment by janelx** [comment]
@greg-does-weather removing the criteria for translation ready makes sense. @colinmurphy01 do you see any reason why we shouldn't pull out translation into its own stand alone issue?

**Comment by colinmurphy01** [comment]
Yea, you're right. I think should can move forward with this component without full translation support and document the translation needs for this component separately based on Greg's comments.

**Comment by mgwalker** [comment]
I created #204 for making this component translation-ready.

**Comment by janelx** [comment]
@greg-does-weather I just realized the shortDescription from Forecast API is not the same a icon api "description". I have it designed to use the shorter icon api "description". Going to confirm w/Shad, but I think it's ok to start with short icon api description. If another we think it's important to see the that shorterDescription from the API, my recommendation is we design for it in a future iteration (detail weather view) after some testing on this.

**Comment by greg-does-weather** [comment]
So we should be using the descriptions from [the icon mapping spreadsheet](https://docs.google.com/spreadsheets/d/1Er4nW_WA2nu-9GdeoUwT4JkSbfp7B84hUnXa32PQcdk/edit#gid=1809319678)? That's doable!

**Comment by colinmurphy01** [comment]
@greg-does-weather small request but important for NWs shareout and onsite. Can you update the location to Nashville instead of Minneapolis?

**Comment by mgwalker** [comment]
![you got it, dude](https://media.giphy.com/media/l41lZxzroU33typuU/giphy.gif?cid=ecf05e47dd7djlywx7gkutzwln3x0jmm7c3awhr1mc98h8bt&ep=v1_gifs_search&rid=giphy.gif&ct=g)

**Comment by colinmurphy01** [comment]
@greg-does-weather what do we need to do to call this done?

Have we done a review with Eng, Design, PO?

**Comment by greg-does-weather** [comment]
I hope PR #331 is the last step. The original implementation was kind of messy and it's been iterating since then: adding translation, moving CSS out, etc. It's in review now with Claire and Janel.

Implementation on this one hasn't been ideal, but I think it's been helpful to me to work through it in pieces.
